tests/test_edf.py

```diff
@@ -1029,3 +1011,84 @@ def test_get_starttime_from_file_with_reserved_field_indicating_edfplus_but_no_a
     edf._reserved = Edf.reserved.encode("EDF+C")
     edf.write(tmp_file)
     assert read_edf(tmp_file).starttime == starttime
+
+
+@pytest.mark.parametrize("selection", [[1], ["EDF Annotations"]])
+def test_drop_signals_does_not_allow_dropping_timekeeping_signal(selection):
+    edf = Edf(
+        signals=[
+            EdfSignal(np.arange(2), sampling_frequency=1),
+            _create_annotations_signal(
+                [EdfAnnotation(0, None, "ann 1")],
+                num_data_records=2,
+                data_record_duration=1,
+            ),
+            _create_annotations_signal(
+                [EdfAnnotation(0.25, None, "ann 2")],
+                num_data_records=2,
+                data_record_duration=1,
+                with_timestamps=False,
+            ),
+        ],
+    )
+    with pytest.raises(ValueError, match="Can not drop EDF\\+ timekeeping signal"):
+        edf.drop_signals(selection)
+
+
+@pytest.mark.parametrize(
+    "new_signal",
+    [
+        EdfSignal(np.arange(100), 10),
+        EdfSignal(np.arange(200), 20),
+        EdfSignal(np.arange(30), 3),
+    ],
+)
+def test_append_signals_single(new_signal: EdfSignal):
+    edf = Edf([EdfSignal(np.arange(100), 10), EdfSignal(np.arange(50), 5)])
+    new_signal = EdfSignal(np.arange(1000), 100)
+    edf.append_signals(new_signal)
+    assert edf.num_signals == 3
+    assert edf.signals[2] == new_signal
+
+
+def test_append_signals_multiple():
+    edf = Edf(
+        [
+            EdfSignal(np.arange(100), 10),
+            EdfSignal(np.arange(50), 5),
+        ]
+    )
+    new_signals = [
+        EdfSignal(np.arange(100), 10),
+        EdfSignal(np.arange(30), 3),
+    ]
+    edf.append_signals(new_signals)
+    assert edf.num_signals == 2 + len(new_signals)
+    for actual, expected in zip(new_signals, edf.signals[2:]):
+        assert actual == expected
+
+
+@pytest.mark.parametrize(
+    ("length", "sampling_frequency"),
+    [(1001, 10), (999, 10), (1000, 1.0001), (1, 0.011)],
```

Comment:
Shouldn't this be like this?
```suggestion
    [(1001, 10), (999, 10), (1000, 10.001), (1, 0.011)],
```

Reply:
Thanks for spotting that, yes! Fixed in f0f9128